pandas/tests/frame/test_combine_concat.py

```diff
@@ -763,3 +763,23 @@ def test_concat_datetime_datetime64_frame(self):
 
         # it works!
         pd.concat([df1, df2_obj])
+
+
+class TestDataFrameUpdate(TestData):
+
+    def test_update_nan(self):
+        # #15593 #15617
+        # test 1
+        expected = df1 = DataFrame({'A': [1.0, 2, 3], 'B': date_range('2000', periods=3)})
+        df2 = DataFrame({'A': [None, 2, 3]})
+        df1.update(df2, overwrite=False)
+
+        tm.assert_frame_equal(df1, expected)
+
+        # test 2
+        expected = df1 = DataFrame({'A': [1, None, 3], 'B': date_range('2000', periods=3)})
```

Comment:
Sorry, I think this got lost in the conversation, but can you write this as:

```python
df1 = DataFrame({'A': [1, None, 3], 'B': date_range('2000', periods=3)})
df2 = DataFrame({'A': [None, 2, 3]})
expected = DataFrame({'A': [1, 2, 3], 'B': date_range('2000', periods=3)})
```

The one on line 773 can be a bit shorter, just 

```python
df1 = DataFrame({'A': [1.0, 2, 3], 'B': date_range('2000', periods=3)})
expected = df1.copy()
```

basically, since `.update` works inplace, the test will always be true no matter what happens. `expected` and `df1` are reference to the same object. It's like

```python
In [2]: a = b = {'x': 1}

In [3]: a['y'] = 2

In [4]: a == b
Out[4]: True
```

Which "shouldn't" be true (but is, since `a` and `b` are both references to the same object. Let me know if that makes sense.